testhelpers/org.eclipse.n4js.tests.helper/src/org/eclipse/n4js/tests/helper/documentprovider/CountPostChangeBroadcastChangeNotificationManager.xtend

```diff
@@ -20,16 +20,20 @@ import org.eclipse.core.resources.IResourceChangeEvent
   * This class counts the number of broadcastChange method calls.
   */
 class CountPostChangeBroadcastChangeNotificationManager extends NotificationManager {
-
+	Thread mainThread
 	int countPostChangeBroadcastTriggered;
 
-	new(Workspace workspace) {
+	new(Workspace workspace, Thread mainThread) {
 		super(workspace)
+		this.mainThread = mainThread
 	}
 
 	override void broadcastChanges(ElementTree lastState, ResourceChangeEvent event, boolean lockTree) {
-		if (event.type == IResourceChangeEvent.POST_CHANGE)
+		// Count the post change events triggered on the main thread
+		if (Thread.currentThread == mainThread && event.type == IResourceChangeEvent.POST_CHANGE) {
 			countPostChangeBroadcastTriggered++
+			println(Thread.currentThread.stackTrace)
```

Comment:
Use logger and `WARN` or `DEBUG` level. Do not pollute the logs if no need.

Reply:
OOps it was for debugging I forget to remove it